### PR TITLE
Simple CI integration with Travis added .travis.yml

### PR DESCRIPTION
**Pull request by fatz**
- `go get`
- diff on `gofmt`
- run `go tool vet`
- run `go test -v -race`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+sudo: false
+language: go
+go:
+  - "1.11.x"
+  - "1.10.x"
+  - "1.9.x"
+  - master
```
Comment (bamarni) [inline .travis.yml]: Is it really needed to test against Go master? We were doing it in the CLI and it slows down builds from quite a lot, as they have to build Go from source.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+matrix:
+  allow_failures:
+    - go: master
+  fast_finish: true
+script:
+  - go get -t -v ./...
+  - diff -u <(echo -n) <(gofmt -d -s .)
+  - go tool vet .
+  - go test -v -race ./...
```
Comment (bamarni) [inline .travis.yml]: Shall we have this in a Makefile?

File: .travis.yml (hunk 1 above)
Comment (fatz) [reply .travis.yml]: So we ignore errors for it. But TBH i didn’t thought about the build time. 

I would be ok with dropping it. Maybe we could consider testing master against it later on

File: .travis.yml (hunk 2 above)
Comment (fatz) [reply .travis.yml]: Yeah makes sense but wanted a quick solution for now. 

Beside Makefile we also should vendor dependencies and Test dependencies. 

I’ve ignored that for now to just getting started with CI asap

Comment (bamarni) [reply .travis.yml]: Sounds good to start with :+1:

File: .travis.yml (hunk 1 above)
Comment (bamarni) [reply .travis.yml]: I'd rather leave it off, Travis builds would just take 1-2 minutes without it. And we only do common usage of the Go http package.

Comment (janisz) [reply .travis.yml]: Do we need all versions?

Comment (fatz) [reply .travis.yml]: will drop master. Not sure if we really need 1.9 but I think the additional time would be fairly low

**Pull request merged**